### ISSUE TITLE
Failed to put role on disk: `Error opening /dev/sda: Permission denied`

### ISSUE THREAD

**Issue by kinvaris** [openvstorage/framework#674]
While trying to put a role on this disk:

```
Traceback (most recent call last):
  File "/usr/lib/python2.7/dist-packages/celery/app/trace.py", line 240, in trace_task
    R = retval = fun(*args, **kwargs)
  File "/usr/lib/python2.7/dist-packages/celery/app/trace.py", line 438, in __protected_call__
    return self.run(*args, **kwargs)
  File "/opt/OpenvStorage/ovs/lib/helpers/decorators.py", line 297, in new_function
    output = function(*args, **kwargs)
  File "/opt/OpenvStorage/ovs/lib/storagerouter.py", line 1933, in configure_disk
    DiskController.sync_with_reality(storagerouter_guid)
  File "/usr/lib/python2.7/dist-packages/celery/local.py", line 167, in <lambda>
    __call__ = lambda x, *a, **kw: x._get_current_object()(*a, **kw)
  File "/usr/lib/python2.7/dist-packages/celery/app/trace.py", line 439, in __protected_call__
    return orig(self, *args, **kwargs)
  File "/usr/lib/python2.7/dist-packages/celery/app/task.py", line 420, in __call__
    return self.run(*args, **kwargs)
  File "/opt/OpenvStorage/ovs/lib/helpers/decorators.py", line 297, in new_function
    output = function(*args, **kwargs)
  File "/opt/OpenvStorage/ovs/lib/disk.py", line 148, in sync_with_reality
    partitions_info = DiskTools.get_partitions_info(match.groups()[0])
  File "/opt/OpenvStorage/ovs/extensions/generic/disk.py", line 101, in get_partitions_info
    raise RuntimeError('{0} {1}'.format(cpe, cpe.output))
RuntimeError: Command 'parted -m /dev/sda unit B print -s' returned non-zero exit status 1 Error: Error opening /dev/sda: Permission denied
```

This command is executed by the `ovs` user, but when executing with `root` it works.

```
ovs@cmp01:~$  parted -m /dev/sda unit B print -s
Error: Error opening /dev/sda: Permission denied

root@cmp01:~# parted -m /dev/sda unit B print -s
BYT;
/dev/sda:90028302336B:scsi:512:512:gpt:ATA KINGSTON SVP200S;
```

**Comment by wimpers** [comment]
@kinvaris which release was this? Is this during automated install? Most likely an issue on the ansible side.  

 This needs to be run as root (not as OVS user).

**Comment by kinvaris** [comment]
No this is not during setup. This is when assigning the roles (DB, SCRUB, WRITE & READ), a task is launched by the framework to format and partition the disk. During one of its subroutines it stops because this command is executed by the user `ovs` and it should be executed by `root`.

**Comment by khenderick** [comment]
The sync is executed as the user `ovs` and there's one path in the sync that needs `root`. This duplicates some internal ticket that's been open for quite some time.

**Comment by wimpers** [comment]
@khenderick @kinvaris how come the assigning roles doesn't always fail in this case?  You would expect that or it works or it doesn't work. What is the scenario when it doesn't work.

**Comment by khenderick** [comment]
@wimpers, the error occurs when there are partitions on the disk, but it can't load the partition number from udev. In that case it fall back to another method with requires root access.

This issue has been reported numerous times and it was always decided to not fix it. But as it keeps being reported and because it's quite an easy fix anyway, it might be a good idea to fix it.

**Comment by khenderick** [comment]
Technical implementation details; We're running in a remote context in this code, so we can just switch to a remote for the DiskTools as well

**Comment by pploegaert** [comment]
indeed, I already have a partly fix for that in code, still needs to be tested

**Comment by khenderick** [comment]
This was apperently fixed by #833 but the ticket's status wasn't updated

**Comment by JeffreyDevloo** [comment]
### Information

We encountered this error during OVH setup. The proposed fix was to execute the parted commands as root.
The code should have been update.

### Steps
- Installation of OVH setup
- Setup disk roles
- Got no issues (when command was executed as root)

### Test result

Test passed.
Issue has been fixed for openvstorage-2.7.3-rev.3964.4c54f7e
Other similar tickets affected by the change: https://github.com/openvstorage/framework/issues/913
